compiler/rustc_builtin_macros/src/autodiff.rs

```diff
@@ -25,6 +25,16 @@ mod llvm_enzyme {
 
     use crate::errors;
 
+    pub(crate) fn outer_normal_attr(
```

Comment:
I thought about putting it into the rustc_ast crate, but I couldn't find any other users and given jana's attribute rework I don't think it makes sense to add extra functions there, especially given someone is looking into moving autodiff over to the new attribute handling infra.